<launch>
    <arg name="file_name"  default="/home/nihao/workspace/autoware.ai/src/autoware/common/map_file/lanelet2_maps_20250422.osm"/>
    <node pkg="map_file" type="lanelet2_map_loader" name="lanelet2_map_loader" output="screen">
        <param name="lanelet2_path" value="$(arg file_name)" />
    </node>

    <!-- 测试时车辆tf位姿发布 -->
    <arg name="x_set" default="47.4" />
    <arg name="y_set" default="8.905" />
    <arg name="yaw_set" default="190" />
    <node name="car_tf_publisher_test" pkg="map_file" type="car_tf_publisher_test" output="screen">
        <param name="yaw_set" value="$(eval yaw_set)" />
        <param name="x_set" value="$(eval x_set)" />
        <param name="y_set" value="$(eval y_set)" />
    </node>

    <!-- 定义车辆长度和宽度参数 -->
    <arg name="length" default="6" />
    <arg name="width" default="2.2" />
    <arg name="draw_flag" default="false" />
    <arg name="rotation_margin" default="15" />


    <!-- 启动距离计算服务节点，并传递计算后的参数 -->
    <node name="corner_distance_publisher" pkg="map_file" type="corner_distance_publisher" output="screen">
        <param name="front_right_x" value="$(eval arg('length') / 2)" />
        <param name="front_right_y" value="$(eval -arg('width') / 2)" />
        <param name="rear_right_x" value="$(eval -arg('length') / 2)" />
        <param name="rear_right_y" value="$(eval -arg('width') / 2)" />
        <param name="rear_left_x" value="$(eval -arg('length') / 2)" />
        <param name="rear_left_y" value="$(eval arg('width') / 2)" />
        <param name="front_left_x" value="$(eval arg('length') / 2)" />
        <param name="front_left_y" value="$(eval arg('width') / 2)" />
        <param name="draw_flag" value="$(arg draw_flag)" />
        <param name="rotation_margin" value="$(arg rotation_margin)" />
    </node>
    
</launch>